File: backend/app/schemas/action_item.py
from typing import Literal

from pydantic import BaseModel


Priority = Literal[
    "High",
    "Medium",
    "Low",
]

Status = Literal[
    "Pending",
    "Completed",
]


class ActionItemCreate(BaseModel):
    title: str
    priority: Priority
    due_date: str | None = None


class ActionItemResponse(BaseModel):
    id: int
    interaction_id: int
    hcp_id: int

    title: str

    priority: Priority

    status: Status

    due_date: str | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/actionItem.ts
export interface Task {
    id: number;
    title: string;
    priority: "High" | "Medium" | "Low";
    status: "Pending" | "Completed";
    due_date?: string | null;
}

export interface HCPActionGroup {
    hcp: {
        id: number;
        name: string;
        hospital: string;
    };

    pending: number;
    completed: number;

    tasks: Task[];
}